src_c/include/base.h

```diff
@@ -188,4 +188,97 @@ pg_UintFromObjIndex(PyObject *obj, int _index, Uint32 *val)
     return result;
 }
 
+// these return PyObject * on success and NULL on failure.
+
+static PG_FORCE_INLINE PyObject *
+pg_TupleFromDoubles(double val1, double val2)
+{
+    /*this is demonstrated to be faster than Py_BuildValue*/
+    PyObject *tuple = PyTuple_New(2);
+    if (!tuple)
+        return NULL;
+
+    PyObject *tmp = PyFloat_FromDouble(val1);
+    if (!tmp) {
+        Py_DECREF(tuple);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(tuple, 0, tmp);
+
+    tmp = PyFloat_FromDouble(val2);
+    if (!tmp) {
+        Py_DECREF(tuple);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(tuple, 1, tmp);
+
+    return tuple;
+}
+
+static PG_FORCE_INLINE PyObject *
+pg_TupleFromInts(int val1, int val2)
+{
+    /*this is demonstrated to be faster than Py_BuildValue*/
+    PyObject *tuple = PyTuple_New(2);
+    if (!tuple)
+        return NULL;
+
+    PyObject *tmp = PyLong_FromLong(val1);
+    if (!tmp) {
+        Py_DECREF(tuple);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(tuple, 0, tmp);
+
+    tmp = PyLong_FromLong(val2);
+    if (!tmp) {
+        Py_DECREF(tuple);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(tuple, 1, tmp);
+
+    return tuple;
+}
+
+static PG_FORCE_INLINE PyObject *
+pg_PointSequence_FromArrayDouble(double *array, int len, int ret_list)
```

Comment:
the name in the function is ok with me, but maybe change the signature a bit to something like this:
`pg_PointSequence_FromArrayDouble(double *array, int length, int tuple_size, int use_tuple)`

Reply:
What should `int tuple_size` do?